blob/fileblob/fileblob.go

```diff
@@ -250,22 +254,34 @@ func (b *bucket) NewTypedWriter(ctx context.Context, key string, contentType str
 		ContentType: contentType,
 		Metadata:    metadata,
 	}
-	return &writer{
+	w := &writer{
 		ctx:   ctx,
 		w:     f,
 		path:  path,
 		attrs: attrs,
-	}, nil
+	}
+	if len(opts.ContentMD5) > 0 {
+		w.contentMD5 = opts.ContentMD5
+		w.md5hash = md5.New()
+	}
+	return w, nil
 }
 
 type writer struct {
-	ctx   context.Context
-	w     io.WriteCloser
-	path  string
-	attrs xattrs
+	ctx        context.Context
+	w          io.WriteCloser
+	path       string
+	attrs      xattrs
+	contentMD5 []byte
+	md5hash    hash.Hash
 }
 
 func (w writer) Write(p []byte) (n int, err error) {
+	if w.md5hash != nil {
+		if _, err := w.md5hash.Write(p); err != nil {
+			return 0, err
```

Comment:
The returned err could probably use some context.

`return 0, fmt.Errorf("updating md5 hash: %v", err)`

Reply:
Done.